source/feature/openvswitch/openvswitch-spec.html.md

```diff
@@ -0,0 +1,183 @@
+
+This aproach is prefered over having an additional configurator type under the existing switch scheme:
+
+- Adding OVS as a configurator may require changes in the upper common levels, which may introduce instability in existing functionality.
+The proposed abstraction level is aimed to leave existing logic isolated from OVS (and vice versa), reducing potential regression and instability.
+- Different switches or domains should be decoupled, each having their own underling implementation options allowing full flexability on how they are implemented.
+- The modeling of each switch is different, having it seperated reduces complex logic.
+
+Switch implementations are registered per availability, with an identification and a callback. 
+setupNetworks requests are expected to arrive from Engine with a switch type ID, based on which the request is forwarded to the switch implementation.
+
+
+### OVS Configurators
+There are at least two ways to configure OVS: Using the command line utilities (i.e ovs-vsctl) and OVSDB protocol python binding. The latter is considered faster.
+To allow future optimizations, an abstraction is defined with the command line as its implementation. This adopts similar work done on openstack ovsdb agent.
+API Ref: https://review.openstack.org/#/c/143709/20/neutron/agent/ovsdb.py
+
+### IP Interface
+Host IP settings are defined on the kernel stack, independent of the network switch type.
+Current Linux-native networking implementation mixes between the two. For OVS, we would like to put things in better order: the IP setup actions are to be organized under the same “roof”, implemented using configurators.
+
+### Persistence
+OVS persistency is embedded in the switch.
+IP settings are not included and may be implemented using the current persistence scheme.
+- Note: An alternative may be to use a minimal persistent management settings and zero persistency for all the rest. Nevertheless these changes are out of scope and are to be handled under a zero persistency feature. 
+
+### Rollback
+VDSM networking uses rollback to recover from a setup request which faild to complete.
+The network API setupNetworks verb has a complex rollback semantics.
+In the vdsm api context, a rollback is detected and a command is issued to supervdsm for handling. This function is currently unused, because the ifcfg configurator never triggers it.
+Handling the rollback as close as possible to the setupNetworks api implementation seems more reasonable, moving this rollback handler to the network api (currently in supervdsm context) is to be considered.
+The OVS rollback is to be integrated into the existing rollback functionality without heavely affecting existing beahaviours:
+
+- OVS configurators should trigger a rollback which is to be handled at the api level (as described above).
+- The ifcfg configurator should keep the existing logic, but in addition it should inform the api level of the rollback which it handled.
```

Comment:
Because the upper level may need to rollback the IP settings or possibly OVS settings (when we migrate from ovs to legacy).